### PR TITLE
Twig ENV and SESSION fix

### PR DESCRIPTION
**Pull request by jamiekohns**
It took me a bit, bug I figured out why ENV and SESSION weren't loading- my bad, I've not used twig outside of a framework!

### CODE REVIEW

File: app/init.php [hunk 1]
```diff
@@ -1,19 +1,31 @@
 <?php
+// Composer autoload
```
Comment (jamiekohns) [inline app/init.php:2]: Add comments!

File: app/init.php [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
 $dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
 $dotenv->load();
 
+// initialize Twig template engine
 $loader = new \Twig\Loader\FilesystemLoader(__DIR__ . '/templates');
 $twigConfig = [];
-if (getenv('PRODUCTION')) {
+
+// If we're in production env, use cache
+if ($_ENV['PRODUCTION'] === true) {
```
Comment (jamiekohns) [inline app/init.php:14]: In session, I kept telling y'all to use `getnev`, but that doesn't work here (it's been _depricated_, or removed).

You were all using the correct form : `$_ENV` to access our `.env` settings!

File: app/init.php [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
     $twigConfig['cache'] = __DIR__ . '/cache';
+} else {
+    // if not production, use debug extension
+    $twigConfig['debug'] = true;
```
Comment (jamiekohns) [inline app/init.php:18]: To help out, I set up the debugging extension in twig- if `PRODUCTION` is `FALSE`, you will be able to use 
```
{{ dump(variable) }}
```
in templates for debugging

File: app/init.php [hunk 4]
```diff
@@ -1,19 +1,31 @@
...
-include_once $_ENV['BASE_DIRECTORY'] . '/web-assets/app_nav/footer.php';
-include_once $_ENV['BASE_DIRECTORY'] . '/web-assets/app_nav/nav.php';
-*/
+// start a session
 session_start();
-?>
\ No newline at end of file
+
+// add our Dotenv variables to the Twig 'ENV' variable
+$twig->addGlobal('ENV', $_ENV);
```
Comment (jamiekohns) [inline app/init.php:28]: Once everything it set up, we have to explicitly tell twig what our `ENV` and `SESSION` variables are. With this, they will be available in templates as `{{ SESSION.user.name }}` etc.

File: app/templates/components/nav.twig [hunk 5]
```diff
@@ -3,44 +3,23 @@
 
         <!-- <img src="/web-assets/images/BDPA-Flights-Black.jpeg" class="rounded" width="44" height="44" alt="" loading="lazy"> -->
 
-        <a href="<?=$_ENV['BASE_URL']?>" class="navbar-brand mb-0 h1 ml-3">Ghostmeme</a> <!-- Make name Airlanta -->
+        <a href="#" class="navbar-brand mb-0 h1 ml-3">Ghostmeme</a> <!-- Make name Airlanta -->
```
Comment (jamiekohns) [inline app/templates/components/nav.twig:6]: Fixme!

File: app/templates/components/nav.twig [hunk 6]
```diff
@@ -3,44 +3,23 @@
...
         <div class="col col-md w-100 justify-content-center">
-            <form method='get' action="<?=$_ENV['BASE_URL']?>">
+            <form method='get' action="/search.php">
```
Comment (jamiekohns) [inline app/templates/components/nav.twig:8]: `search.php` doesn't exist, this is just for the twig example- another "fixme"

File: app/templates/components/nav.twig [hunk 7]
```diff
@@ -3,44 +3,23 @@
...
                 <input name='search' type='search' placeholder='search'>
-                <!-- <button class = 'btn btn-primary'type="submit"> Search </button> -->
+                <button class = 'btn btn-primary'type="submit"> Search </button>
             </form>
         </div>
             <div class="navbar-nav ml-auto">
-            
+                {% if not SESSION.user %}
```
Comment (jamiekohns) [inline app/templates/components/nav.twig:14]: Here's our template logic, using the `SESSION` variable

File: app/templates/components/nav.twig [hunk 8]
```diff
@@ -3,43 +3,21 @@
...
-                            break;
-                        case 2:
-                            $showSideBar = true;
-                            ?>
-                                 <a class="nav-item nav-link" href=""> Notifications </a>
-                                 <a class="nav-item nav-link" href="<?= $_ENV['BASE_URL'] . '/logout.php' ?>"> Log Out </a>
-                            <?php
-                            break;
-                    }?>
+                    Hello {{ SESSION.user.name }}
```
Comment (jamiekohns) [inline app/templates/components/nav.twig:19]: This will need a logout link :hint:

File: app/templates/components/nav.twig (hunk 6 above)
Comment (NewTigerShark10) [reply app/templates/components/nav.twig:8]: Right, I did the searching in index.php. I think I'll make a search.php to handle the searching. I don't know why I did it in index having a separate file makes more sense.

File: app/templates/components/nav.twig (hunk 5 above)
Comment (NewTigerShark10) [reply app/templates/components/nav.twig:6]: <a href="index.php" class="navbar-brand mb-0 h1 ml-3">Ghostmeme</a>

I think?

File: app/templates/components/nav.twig (hunk 8 above)
Comment (NewTigerShark10) [reply app/templates/components/nav.twig:19]: `<a class="nav-item nav-link" href="logout.php">Logout</a>`

**Review by NewTigerShark10 — APPROVED** [review]

**Review by NewTigerShark10 — CHANGES_REQUESTED** [review]
the searchbar goes to `https://suchit.hscc-atlanta.com/search.php?search=test1` instead of `https://suchit.hscc-atlanta.com/Ghostmeme/app/search.php?search=test1`, fixed it by changing `  <form method='get' action="/search.php">` to ` <form method='get' action="search.php">`

**Comment by NewTigerShark10** [comment]
We also need a way for the sidenav to be included (`Ghostmeme/app/web-assets/app_nav/sidenav.php`

**Pull request merged**